### PR TITLE
Extract duplicated hasTransactionHash, introduce MineableStatus

### PR DESCRIPTION
**Pull request by rzadp**

### CODE REVIEW

File: universal-login-sdk/lib/core/services/MineableFactory.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import {MineableState} from '@universal-login/commons';
+import {MineableStatus} from '@universal-login/commons/lib/core/models/message';
```
Comment (xitronix) [inline universal-login-sdk/lib/core/services/MineableFactory.ts]: Import from '@universal-login/commons'

Comment (rzadp) [reply universal-login-sdk/lib/core/services/MineableFactory.ts]: 👍 Done

File: universal-login-sdk/lib/core/services/MineableFactory.ts [hunk 2]
```diff
@@ -12,4 +13,10 @@ export class MineableFactory {
   protected isMined(state: MineableState) {
     return state === 'Error' || state === 'Success';
   }
+
+  protected hasTransactionHash(status: MineableStatus) {
+    return ['Pending', 'Success', 'Error'].includes(status.state) &&
+      status.transactionHash !== null &&
+      status.transactionHash !== undefined;
```
Comment (xitronix) [inline universal-login-sdk/lib/core/services/MineableFactory.ts]: Maybe if transaction hash is missing in this case (status state is Pending, Success or Error) throw error? (ensureNotNull)

Comment (rzadp) [reply universal-login-sdk/lib/core/services/MineableFactory.ts]: Done.

File: universal-login-sdk/lib/core/services/MineableFactory.ts [hunk 3]
```diff
@@ -12,4 +13,12 @@ export class MineableFactory {
   protected isMined(state: MineableState) {
     return state === 'Error' || state === 'Success';
   }
+
+  protected hasTransactionHash(status: MineableStatus) {
+    if (['Pending', 'Success', 'Error'].includes(status.state)) {
+      ensureNotNull(status.transactionHash, TransactionHashNotFound);
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/lib/core/services/MineableFactory.ts]: We should add a test for this situation

Comment (rzadp) [reply universal-login-sdk/lib/core/services/MineableFactory.ts]: Done.

File: universal-login-sdk/test/unit/core/ExecutionFactory.ts [hunk 4]
```diff
@@ -99,6 +99,19 @@ describe('UNIT: ExecutionFactory', async () => {
       expect(messageStatus).to.be.deep.eq(status);
       expect(getStatus.callCount).be.eq(2);
     });
+
+    it('should throw when a status should have transaction hash but does not have it', async () => {
+      getStatus.onCall(1).returns({...defaultStatus, state: 'Success', transactionHash: undefined});
+      const execution = await executionFactory.createExecution(signedMessage);
+      await expect(execution.waitForTransactionHash()).to.be.eventually.rejectedWith('Transaction hash is not found in Message Status');
+    });
```
Comment (xitronix) [inline universal-login-sdk/test/unit/core/ExecutionFactory.ts]:
```ts
for state of ['Success', 'Error', 'Pending'] {
  it(`......status${(state)}`) {
    .....state,...
  } #nextpr
}
```

Comment (rzadp) [reply universal-login-sdk/test/unit/core/ExecutionFactory.ts]: OK

Comment (rzadp) [reply universal-login-sdk/test/unit/core/ExecutionFactory.ts]: https://github.com/UniversalLogin/UniversalLoginSDK/pull/1400

**Pull request merged**